docs/docs/tutorials/cypress.md

```diff
@@ -0,0 +1,263 @@
+        });
+    });
+})
+
+```
+
+Note we've made two other changes here:
+
+* we've added the size parameter to the describe string of our spec; and
+* we've modified our `beforeEach` to call `cy.viewport(size)`.
+
+This results in:
+
+[![cypress running test multiple viewports](images/cypress-running-test.png)](images/cypress-running-test.png)
+
+Sweet! Note that the test names now include the device preset name. To see more viewport aliases you can add, checkout 
+[this page](https://docs.cypress.io/api/commands/viewport.html#Arguments).
+
+## Testing functionality that requires authentication
+
+Frequently, we need to create a test account, log in as that user, and interact with some functionality only available to authenticated
+users. To help with this, we've created a custom cypress command `cy.createRandomAccountAndLogin()`. As an example of it in use,
+we''ll be adding another test case to the `mobile_navigation.js` file:
+
+```js
+        it(`should show mobile menu on dashboard`, () => {
+            cy.createRandomAccountAndLogin();
+
+            cy.get('[data-test=mobile-menu]').should('not.be.visible')
+            cy.get('[data-test=menu-button').click()
+            cy.get('[data-test=mobile-menu]').should('be.visible')
+        });
+```
+
+The navbar used on these pages has been given the same attributes that we used before: `data-test="mobile-menu"` and `data-test="menu-button"`.
+
+If you step through this test, you'll see that it makes some `POST` requests to create the account then navigates to `/dashboard`.
+
+[![cypress authenticated test with expanded test](images/cypress-authenticated.png)](images/cypress-authenticated.png)
+
+This is much preferred to working through these flows as a user would. Doing so slows tests down and couples the underlying 
+logic of authentication too closely to the UI for tests that don't really care what the auth UI looks like.
+
+## Running the test suite in JUnit
+
+While the Cypress test runner is a great tool, we don't really want to have to open it every time we make a change. Instead,
+we probably just want these tests to run as a part of our normal test suite. Inspired by [this blog post](https://www.wimdeblauwe.com/blog/2019/2019-06-15-running-cypress-tests-with-testcontainers-for-a-spring-boot-with-thymeleaf-application/),
+we've written a JUnit test class that will run our entire suite.
+
+> Note: the author of the blog post above has also [released a library](https://github.com/wimdeblauwe/testcontainers-cypress/tree/testcontainers-cypress-0.4.0) 
+> to do a lot of the work for us. We're working on integrating this with Springboard soon.
+
+
+You can find this test at `src/test/kotlin/org/springboard/CypressIntegrationTest.kt`
+
+This test will also be run as a part of a `gradle build`, making our CI runs super easy. 
+
+#### Caveat
+A fairly natural next step here might be to write test setup code (e.g. create used account) in the JUnit test. We don't recommend this.
```

Comment:
This Caveat is interesting because it almost directly contradicts my plans for Member Magic. I feel very strongly that we DO need to have some test setup code in order to get accounts into a specific state for testing the UI.

I want to write some Junit based DB setup that builds an account with certain receivable and membership states for Member Magic- and doing that in cypress only would be very clunky or impossible because we do not expose that functionality in the UI at all. 

So- if you need to setup an account for testing in a specific state that cannot be achieved through a series of UI steps, then I believe that setting up the DB state SHOULD be done in Junit test setup code.

Reply:
I do have an idea for this for the future. Essentially, I think that we should have a setup that creates interesting data for test, dev, review apps and staging. 

In this way, we can write interesting setup without worrying about whether it messes with cypress - it will just work.